components/heap/third_party/dartino/cmpctmalloc.c

```diff
@@ -1355,6 +1370,7 @@ size_t cmpct_minimum_free_size_impl(cmpct_heap_t *heap)
 // Called with the lock.
 IRAM_ATTR static void *page_alloc(cmpct_heap_t *heap, intptr_t pages, uintptr_t alignment, void *tag)
 {
+    if (pages == 0) return NULL;  // Size overflow.
```

Comment:
```suggestion
    // If pages == 0, then we assume that the caller ran into an integer
    // overflow. This can happen if PAGES_FOR_BYTES was used on
    // a really big size.
    if (pages == 0) return NULL;
```